packages/core/_src/stm/TMap/definition.ts

```diff
@@ -0,0 +1,32 @@
+export const TMapSym = Symbol.for("@effect/core/stm/TMap")
+export type TMapSym = typeof TMapSym
+
+export const _K = Symbol.for("@effect/core/stm/TMap/K")
+export type _K = typeof _K
+
+export const _V = Symbol.for("@effect/core/stm/TMap/V")
+export type _V = typeof _V
+
+/**
+ * @tsplus type ets/TMap
+ */
+export interface TMap<K, V> {
```

Comment:
Depending on what you prefer, you could make `TMap` extend `Collection`, which gives you access to `Collection` methods in TS+.

You just then have to define a `Symbol.iterator` for the internal map representation.

See example: https://github.com/Effect-TS/core/blob/2128087016bd77d5ca8ce0709cd60b709d707f80/packages/core/_src/stm/TArray/definition.ts#L10-L13

Reply:
hmm... I just realised that I am not sure if this will work. The internal representation of the key value tuples are inside a `TRef`, which means I need a STM to access them, even when using `unsafeGet`. How would I go about doing this?